source/js/storage.js

```diff
@@ -0,0 +1,20 @@
+class NoOpStorage {
+  clear() {}
+  getItem() {}
+  key() {}
+  removeItem() {}
+  setItem() {}
+}
+
+export default function getStorage(type) {
+  let storage;
```

Comment:
Don't check whether things exist using try/catach. Save those for when you literally can't be sure whether an exception will occur (e.g. JSON.parse). Instead, check whether `window` exists, and if it does we know its localStorage/sessionStorage can be used just fine:
```
const haveWindow = (typeof window !== "undefined");

const Storage = {
  localStorage: haveWindow ? window.localStorage : new NoopStorage(),
  sessionStorage: haveWindow ? window.sessionStorage : new NoopStorage()
}

export default Storage;
```
Also note this is explicit about the type of storage we want, which means that static code analysis becomes a lot easier, because there's no ambiguity in the import anymore: if we want sessionStorage, we extract literally that.